types/src/front/resource_permissions.ts

```diff
@@ -0,0 +1,68 @@
+import { ModelId } from "../shared/model_id";
+import { RoleType } from "./user";
+
+// Supported operations for resource permissions.
+export const SUPPORTED_OPERATIONS = ["admin", "read", "write"] as const;
+
+export type PermissionType = (typeof SUPPORTED_OPERATIONS)[number];
+
+/**
+ * Represents permissions assigned to a specific group.
+ *
+ * @property id - Unique identifier for the group (ModelId type)
+ * @property permissions - Array of permissions granted to the group
+ */
+type GroupPermission = {
+  id: ModelId;
+  permissions: PermissionType[];
+};
+
+/**
+ * Represents permissions assigned to a specific role.
+ *
+ * @property role - The type of role (RoleType)
+ * @property permissions - Array of permissions granted to the role
+ */
+type RolePermission = {
+  role: RoleType;
+  permissions: PermissionType[];
+};
+
+/**
+ * Defines group-based permissions for a resource.
+ * Used when access control is managed through group assignments.
+ */
+export type GroupResourcePermissions = {
+  groups: GroupPermission[];
+};
+
+/**
+ * Defines combined group and role-based permissions for a resource.
+ */
+export type CombinedResourcePermissions = {
+  groups: GroupPermission[];
+  roles: RolePermission[];
+  workspaceId: ModelId;
+};
+
+/**
+ * Represents the complete permission configuration for a resource.
+ * Can be either:
+ * - Group-based permissions only
+ * - Both group and role-based permissions combined
```

Comment:
For now we don't need to support role-based permissions only. Adding it will create a lot of optional code in other places (conversations and agent_configurations). Will add it once we've refactored the way we store permissions on those tables.